File: skellig-junit-runner/src/test/java/org/skellig/runner/config/TestSkelligContext.java
package org.skellig.runner.config;

import org.skellig.teststep.processing.converter.TestDataConverter;
import org.skellig.teststep.runner.context.SkelligTestContext;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class TestSkelligContext extends SkelligTestContext {

    @Override
    protected List<TestDataConverter> getAdditionalTestDataConverters() {
        return Collections.singletonList(new CustomMessageTestDataConverter());
    }

    @Override
    protected List<TestStepProcessorDetails> getTestStepProcessors() {
        return Stream.of(
                new TestStepProcessorDetails(
                        new SimpleMessageTestStepProcessor.Builder()
                        .withTestScenarioState(getTestScenarioState())
                        .withValidator(getTestStepResultValidator())
                        .build(),
                        createTestStepFactoryFrom(SimpleMessageTestStepFactory::new)
                )
        ).collect(Collectors.toList());
    }
}

File: skellig-junit-runner/src/test/kotlin/org/skellig/runner/config/TestSkelligContext.kt
package org.skellig.runner.config

import org.skellig.teststep.runner.context.SkelligTestContext

class TestSkelligContext : SkelligTestContext() {

    override val testStepProcessors: List<TestStepProcessorDetails>
        get() = listOf(
            createTestStepProcessorFrom(
                    SimpleMessageTestStepProcessor.Builder()
                        .withTestScenarioState(getTestScenarioState())
                        .withValidator(getTestStepResultValidator())
                        .build()
            ) { testStepRegistry, props, converter -> SimpleMessageTestStepFactory(testStepRegistry, props, converter) },
            createTestStepProcessorFrom(
                    SimpleTestStepProcessor.Builder()
                        .withTestScenarioState(getTestScenarioState())
                        .withValidator(getTestStepResultValidator())
                        .build()
            ) { testStepRegistry, props, converter -> SimpleTestStepFactory(testStepRegistry, props, converter) }
        )
}